znai-tests/src/test/groovy/scenarios/preview.groovy

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2021 znai maintainers
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package scenarios
+
+import data.FsLocations
+
+import static org.testingisdocumenting.webtau.WebTauGroovyDsl.*
+import static pages.Pages.*
+
+scenario('open preview') {
+    previewServer.openPreview()
+}
+
+scenario('footer should be updated on footer file change') {
+    standardView.footer.should == ~/Contributions are welcome/
+
+    fs.writeText(FsLocations.resolveFromZnaiDocs("footer.md"), "new footer")
```

Comment:
changing footer during test and detect preview mode reflects it
@evinaik

Reply:
interesting, looks a lot like react-testing-library+jest (for UI tests). excited to try it!